scripts/scil_visualize_scatterplot.py

```diff
@@ -0,0 +1,215 @@
+                      help='Use the provided info for the title name. '
+                           ' [%(default)s]')
+    scat.add_argument('--x_label', default='x',
+                      help='Use the provided info for the x axis name. '
+                           ' [%(default)s]')
+    scat.add_argument('--y_label', default='y',
+                      help='Use the provided info for the y axis name. '
+                           ' [%(default)s]')
+    scat.add_argument('--label', default='Mask',
+                      help='Use the provided info for the legend box '
+                           'corresponding to mask or probability map. '
+                           ' [%(default)s]')
+    scat.add_argument('--label_prob_2', default=' ',
+                      help='Use the provided info for the legend box '
+                           'corresponding to the second probability map. '
+                           ' [%(default)s]')
+    scat.add_argument('--marker', default='.',
+                      help='Use the provided info for the marker shape.'
+                           ' [%(default)s]')
+    scat.add_argument('--marker_size', default='15',
+                      help='Use the provided info for the marker shape.'
+                           ' [%(default)s]')
+    scat.add_argument('--transparency', default='0.4',
+                      help='Use the provided info for the marker shape.'
+                           ' [%(default)s]')
+    scat.add_argument('--dpi', default='300',
+                      help='Use the provided info for the dpi resolution.'
+                           ' [%(default)s]')
+    scat.add_argument('--color_prob', nargs=2, metavar=('color1', 'color2'),
+                      default=('r', 'b'))
+
+    p.add_argument('--show_only', action='store_true',
+                   help='Do not save the figure, only display.')
+
+    add_overwrite_arg(p)
+
+    return p
+
+
+def load_data(images_list, mask=False, label=False):
+    map = []
+    for curr_map in images_list:
+        load_image = nib.load(curr_map)
+
+        if label:
+            map.append(get_data_as_label(load_image))
+            with open(args.in_atlas_lut) as f:
+                label_dict = json.load(f)
+            return map, zip(*label_dict.items())
+
+        if mask:
+            map.append(get_data_as_mask(load_image))
+        else:
+            map.append(load_image.get_fdata(dtype=np.float32))
+
+    return map
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_x_map, args.in_y_map)
+    if args.probability:
+        assert_inputs_exist(parser, args.prob_mask, args.prob_mask_2)
+
+    if args.in_atlas:
+        assert_inputs_exist(parser, args.in_atlas_lut)
+
+    # Load images
+    maps_image = [args.in_x_map, args.in_y_map]
+    maps_data = load_data(maps_image)
+
+    # Load and apply threshold from mask
+    if args.mask:
```

Comment:
I think you should make sure that mask and probability are not used together I think it doesn't make sense.
I also don't thing the probability option if you choose probability_mask I think it's enough to know what the user wants to do.

Reply:
You need different groups of parameters that will be exclusive.